### DISCUSSION TITLE
Filtering rules for the same collection.

### DISCUSSION THREAD

**Discussion by suelitonx** [pocketbase/pocketbase#4553] [Q&A] [answered]
Greetings, I'm here to ask for help on filtering rules for the same collection. I really appreciate anyone who can help, and sorry for my English.

I have 3 collections:

1. **users** (authentication collection)

2. **locations**
- name
- description

3. **permissions**
- local (relationship with **locations** collection)
- user (relationship with the **users** collection)
- role (permission: 0, 1 or 2)

I want to create a listing rule for the "**locations**" collection, where the user can only list the locations if he has permission to access it (that is, when the permission first exists, and also if it is greater than 0).

I tried the following code, but I was unsuccessful:

`@collection.permissions.user  ?= @request.auth.id &&
(@collection.permissions.local.id ?= id && ( @collection.permissions.user.id = @request.auth.id && @collection.permissions.role > 0 ))`

**Comment by ganigeorgiev** [comment, marked as answer]
The below rule should suffice:

```js
@request.auth.id != "" &&
@collection.permissions.local ?= id &&
@collection.permissions.user ?= @request.auth.id &&
@collection.permissions.role ?> 0 // note that if role is a "select" field the value is always a string, not a number; if that's the case I recommend using named roles
```

If you use PocketBase v0.22.0+ a simpler alternative to the `@collection.*` field is to use [back-relation references](https://pocketbase.io/docs/working-with-relations/#back-relations) (_check also the [caveats](https://pocketbase.io/docs/working-with-relations/#back-relation-caveats) at the bottom_):

```js
@request.auth.id != "" &&
@request.auth.permissions_via_user.local ?= id &&
@request.auth.permissions_via_user.role ?> 0 // note that if role is a "select" field the value is always a string, not a number; if that's the case I recommend using named roles
```

**Reply by suelitonx** [reply]
Even though I used the most current version of Pocketbase, the second option didn't work. However, the first one worked perfectly. Thanks a lot for the help! I'm from Brazil, and I say, Brazil loves you.

**Reply by ganigeorgiev** [reply]
> Even though I used the most current version of Pocketbase, the second option didn't work. 

Both options should produce similar results (unless I misunderstood your collections structure or there is a bug somewhere but from my quick local test I'm not able to reproduce it).

_Note that you can also start the executable with `--dev` to inspect the executed SQL statements. If there is a rule or filter error, you can also find a more detailed error log in the _Admin UI > Logs_ section._
